network-api/networkapi/reports/views.py

```diff
@@ -93,3 +94,42 @@ def get_queryset(self):
         queryset = queryset.order_by("-count", "app_label", "model")
 
         return queryset
+
+
+class BlockTypesReportView(ReportView):
+    title = "Block types report"
+    template_name = "pages/reports/block_types_report.html"
+    header_icon = "placeholder"
+
+    def decorate_paginated_queryset(self, object_list):
+        # Build a cache map of PageBlock's block name to content types
+        page_blocks = PageBlock.objects.all().prefetch_related("page__content_type")
+        page_blocks_to_content_types = {pb.block: [] for pb in page_blocks}
```

Comment:
You could use defaultdict ([see example](https://docs.python.org/3/library/collections.html#defaultdict-examples)) to skip the dictionary comprehension.

e.g. 

```
from collections import defaultdict
...

page_blocks_to_content_types = defaultdict(list)
```

Reply:
Thanks! TIL :mortar_board: